src/docs/release/breaking-changes/use-maxLengthEnforcement-instead-of-maxLengthEnforced.md

```diff
@@ -0,0 +1,208 @@
+* The default value of the maxLengthEnforced property is inferred based on the
+  target platform, to improve the input experience:
+  * Android, Windows: defaults to `MaxLengthEnforcement.enforced`.
+  * iOS, Web, MacOS, Linux, Fuchsia: defaults to
+    `MaxLengthEnforcement.truncateAfterCompositionEnds`.
+
+## Migration guide
+
+The following tips describe how to produce common behaviors, and which behavior
+is using on different platforms by default. Notice that the enforcement API is
+exposed to `TextField`, `TextFormField`, `CupertinoTextField` and
+`LengthLimitingTextInputFormatter`.
+
+_Use the default behavior for the current platform is recommended, since it
+won't break someone that is not surposed to input like the behavior that
+developers has defined._
+
+### Platforms behavior
+
+* Android, Windows: `MaxLengthEnforcement.enforced`. The native behavior of
+  these platforms is enforced. The composing will be handled by the IME while
+  users are entering CJK characters.
+* iOS: `MaxLengthEnforcement.truncateAfterCompositionEnds`. iOS has no default
+  behavior and it requires users implement the behavior themselves. Allow the
+  composition to exceed to avoid breaking CJK input.
+* Web, MacOS, Linux, Fuchsia: `MaxLengthEnforcement.truncateAfterCompositionEnds`.
+  These platforms allow the composition to exceed by default.
+
+### To enforce the limit all the time
+
+To enforce the limit that always truncate the value when it reaches the limit,
+use `MaxLengthEnforcement.enforced` in editable text fields.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+TextField(maxLength: 6)
+```
+
+or 
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforced: true,
+)
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforcement: MaxLengthEnforcement.enforced,
+)
+```
+
+### To not enforce the limitation
+
+To show a max length error in `TextField`, but _not_ truncate when the limit
+is exceed, use `MaxLengthEnforcement.none` instead of
+`maxLengthEnforced: false`.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforced: false,
+)
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforcement: MaxLengthEnforcement.none,
+)
+```
+
+For `CupertinoTextField`, which isn't able to show an error message,
+just don't set the `maxLength` value.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+CupertinoTextField(
+  maxLength: 6,
+  maxLengthEnforced: false,
+)
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+CupertinoTextField(
+  // Leave it not set.
+)
+```
+
+### To enforce the limit, but not for composing text
+
+To avoid truncating text while the user is editing, specify
+`MaxLengthEnforcement.truncateAfterCompositionEnds`. This behavior allows some
+Chinese, Japanese, and Korean (CJK) characters to temporarily ignore the limit
+until editing is complete.
+
```

Comment:
I think the Gboard Android case is worth mentioning, It is going to be a common pitfall.